docs/examples/te_llama/tutorial_accelerate_hf_llama_with_te.ipynb

```diff
@@ -0,0 +1,968 @@
+    "2. Hugging Face's `LlamaModel`\n",
+    "    - Hugging Face's `LlamaDecoderLayer`\n",
+    "3. [Baseline] Running HF `LlamaModel` (Precision: `BF16`)\n",
+    "4. [Improvement 1] Replace `nn.Linear` with TE's `Linear` layers (Precision: FP8)\n",
+    "    - Understanding the aberration with Improvement 2 (CPU Overheads)\n",
+    "5. [Improvement 2] Replace HF's `LlamaDecoderLayer` with TE's `TransformerLayer` (Precision: `BF16`)\n",
+    "    - TransformerEngine's `TransformerLayer`\n",
+    "    - `TransformerLayer` options explained\n",
+    "    - Comparing Hugging Face's `LlamaDecoderLayer` with TranformerEngine's `TransformerLayer`\n",
+    "    - Mapping weights from HF's `LlamaDecoderLayer` to TE's `TransformerLayer`\n",
+    "6. [Improvement 3] Replace HF's `LlamaDecoderLayer` with TE's `TransformerLayer` (Precision: `FP8`)\n",
+    "7. Benchmarks revisited and Conclusion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "82113cd0",
+   "metadata": {},
+   "source": [
+    "## From \"Transformer\" to \"Llama\" \n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"media/transformer_llama.png\" width=\"50%\">\n",
+    "    <figcaption> Fig 1: Llama visualized as a transformer. (generated with <a href=\"https://catalog.ngc.nvidia.com/orgs/nvidia/teams/ai-foundation/models/sdxl\">Nvidia's AI-foundation models</a>)</figcaption>\n",
+    "</figure>\n",
+    "\n",
+    "A flashback:\n",
+    "- 2017: [\"Attention Is All You Need\"](https://arxiv.org/abs/1706.03762) paper introduced pioneering \"Transformer\" architecture and changed the NLP field forever.\n",
+    "- 2018-2020: Emergence of GPT model series that showed causal decoder architectures are great fit for pretraining, few-shot and zero-shot learning.\n",
+    "- Fast forward to 2023-2024: Following GPT-3/GPT-4 success stories, researchers and companies raced to produce the next best pretrained model that could further be finetuned for application-specific use-cases. \n",
+    "- One of the latest in this line of pretrained models which is also open source is Meta's [Llama 2](https://llama.meta.com/llama2) models (Large Language Model Meta AI). \n",
+    "    - These models range from 7B to 65B parameters.\n",
+    "    - LLaMA 2 was pretrained on 2 trillion tokens.\n",
+    "\n",
+    "For more information on Llama 2 consider reading the [Huggingface tutorial](https://huggingface.co/blog/llama2). As a quick summary, here are some of the important differences b/w the conventional transformer decoder architecture vs Llama2 architecture:\n",
+    "\n",
+    "1. Decoder only model (causal language modeling and next word prediction)\n",
+    "2. RMSNorm in place of the LayerNorm\n",
+    "3. SwiGLU activation function\n",
+    "4. RoPE as positional embeddings \n",
+    "5. Grouped Query Attention\n",
+    "6. Trained on 4K context length\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"media/transformers_vs_llama.png\" width=\"100%\">\n",
```

Comment:
TBH this image looks kind of lame. On the left side you have an image that you took from the web and it looks blurry. On the right side you have your own picture where you wanted to match the style of the other one, which is nice, but it does not look quite right, e.g. plus signs in the circles are off center, the waves in the other circles look different compared to the original. Also the "N x" text is very big and jarring compared to much smaller version of it on the left.

Suggestion - please redo the left hand side picture to just show the decoder (so GPT basically - that will be much closer to the Llama picture), please fix the symbols and change "N x" to "N×" (use the unicode character for times rather than "x" letter)

Reply:
And 1 more thing - why do you mention KV-cache here? It is an inference optimization, not something Llama-specific.

Reply:
Right, taking a screen shot twice decreased the resolution of the picture on the left. I've now recreated the diagram